common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/YubiKitSmartcardSession.java

```diff
@@ -0,0 +1,174 @@
+     * @throws BadResponseException in case of incorrect YubiKey response
+     * @throws IOException in case of connection error
+     */
+    @NonNull
+    @Override
+    public List<ICertDetails> getCertDetailsList() throws ApduException, BadResponseException, IOException {
+        //Create ArrayList that contains cert details only pertinent to the cert picker.
+        final List<ICertDetails> certList = new ArrayList<>();
+        //We need to check all four PIV slots.
+        //AUTHENTICATION (9A)
+        getAndPutCertDetailsInList(AUTHENTICATION, piv, certList);
+        //SIGNATURE (9C)
+        getAndPutCertDetailsInList(SIGNATURE, piv, certList);
+        //KEY_MANAGEMENT (9D)
+        getAndPutCertDetailsInList(KEY_MANAGEMENT, piv, certList);
+        //CARD_AUTH (9E)
+        getAndPutCertDetailsInList(CARD_AUTH, piv, certList);
+        return certList;
+    }
+
+    /**
+     * Helper method that handles reading certificates off YubiKey, generating YubiKitCertDetails, and inserting into list.
+     * @param slot A PIV slot from which to read the certificate.
+     * @param piv A PivSession created from a SmartCardConnection that can interact with certificates located in the PIV slots on the YubiKey.
+     * @param certList A List collecting the YubiKitCertDetails of the certificates found on the YubiKey.
+     * @throws IOException          in case of connection error
+     * @throws ApduException        in case of an error response from the YubiKey
+     * @throws BadResponseException in case of incorrect YubiKey response
+     */
+    private void getAndPutCertDetailsInList(@NonNull final Slot slot,
+                                            @NonNull final PivSession piv,
+                                            @NonNull final List<ICertDetails> certList)
+            throws IOException, ApduException, BadResponseException {
+        final String methodTag = TAG + ":getAndPutCertDetailsInList";
+        try {
+            final X509Certificate cert =  piv.getCertificate(slot);
+            //If there are no exceptions, add this cert to our certList.
+            certList.add(new YubiKitCertDetails(cert, slot));
+        } catch (final ApduException e) {
+            //If sw is 0x6a82 (27266), This is a FILE_NOT_FOUND error, which we should ignore since this means the slot is merely empty.
+            if (e.getSw() == 0x6a82) {
+                Logger.verbose(methodTag, slot + " slot is empty.");
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Verifies PIN attempt provided by user.
+     * @param pin Char array with pin.
+     * @return true if PIN is verified; false otherwise.
+     * @throws ApduException in case of an error response from the YubiKey
+     * @throws IOException in case of connection error
+     */
+    @Override
+    public boolean verifyPin(char[] pin) throws ApduException, IOException {
```

Comment:
Not seeing any `null` check here, not sure if it can be -- maybe you want to be defensive on sanitizing the input

Reply:
Adding a null check for a EditText component that could lead to an error, and verified that `pin` is @Nonnull

Reply:
I followed the path to get to the pin array here, which included finding that EditText could be null (and handling the null check at that point), and I _think_ now the Pin array shouldn't ever be null. 
Right now there isn't specific logic to treat an empty pin array any differently than a filled one. I'm not sure if it's possible to actually set an empty PIN, but I'm thinking it's probably highly discouraged :) 
An enhancement I can make in a future PR is to disable the pin AlertDialog positive button when the EditText is empty so that the user can never submit an empty PIN.